lib/grovepi/i2c/state.ex

```diff
@@ -1,37 +1,66 @@
 defmodule GrovePi.I2C.State do
+  @moduledoc false
+
   alias __MODULE__
+
   defstruct responses: [], writes: []
-  @moduledoc false
 
-  def add_input(%State{} = state, message) do
+  def add_input(%State{} = state, write) do
     %{state |
-      writes: [add_time_to_messages(message) | state.writes]
+      writes: [add_time_to_write(write) | state.writes]
     }
   end
 
   def add_responses(%State{} = state, responses) do
     %{state | responses: state.responses ++ responses}
   end
 
+  def pop_all_writes(%State{writes: []} = state) do
+    {{:error, :no_more_messages}, state}
+  end
+
+  def pop_all_writes(%State{} = state) do
+    {all_writes, new_state} = get_all_writes(state)
```

Comment:
I think we should remove the intermediate state here.  I would even get rid of the 'get_all_writes' and only have this function.

Reply:
I considered leaving a note on this function.  I thought about removing the intermediate state since `Map.get_and_update_state/3` returns a tuple in the form of the expected return. I decided to leave it to be more explicit about what the function returns.  I can make the update you mentioned in a future pull request.